### PR TITLE
Updating both implementations to use FluentIterables instead of creat…

### PR DESCRIPTION
**Pull request by rpmoore**
…ing new arrays to prevent extra data copying.

### CODE REVIEW

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/csv/DetailedObjectsPhysicalView.java [hunk 1]
```diff
@@ -15,63 +15,116 @@
...
 import com.spectralogic.ds3client.models.*;
 import com.spectralogic.ds3client.utils.Guard;
 
-import java.util.ArrayList;
+import javax.annotation.Nullable;
 
 import static com.spectralogic.ds3cli.util.Utils.*;
+import static com.spectralogic.ds3cli.views.csv.CsvView.DATE_FORMAT;
 
-public class DetailedObjectsPhysicalView extends CsvView<GetDetailedObjectsResult> {
```
Comment (berkeleysquare) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/csv/DetailedObjectsPhysicalView.java]: CsvView could be deleted.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/csv/DetailedObjectsPhysicalView.java [hunk 2]
```diff
@@ -15,63 +15,116 @@
...
-        initTable(ImmutableList.of("Name", "Bucket", "Owner", "Size", "Type", "Creation Date", "Barcode", "State"));
+        final ImmutableList<String> headers = ImmutableList.of("Name", "Bucket", "Owner", "Size", "Type", "Creation Date", "Barcode", "State");
 
-        return renderTable();
+        final FluentIterable<DetailedTapeInfo> objects = FluentIterable.from(obj.getObjIterator())
+                .filter(new Predicate<DetailedS3Object>() {
+                    @Override
+                    public boolean apply(@Nullable final DetailedS3Object input) {
+                        return input.getBlobs() != null && !Guard.isNullOrEmpty(input.getBlobs().getObjects());
+                    }
```
Comment (GraciesPadre) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/csv/DetailedObjectsPhysicalView.java]: Maybe a method on DetailedS3Object to have it indicate whether it has stuff to process?

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/csv/DetailedObjectsPhysicalView.java]: That feels like we'd be adding behavior into a pojo.

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/csv/DetailedObjectsPhysicalView.java]: What we could do is instead pull that functionality into its own method inside of DetailedObjectsPhysicalView

**Pull request merged**